### PR TITLE
Add ARK RTK GPS

### PR DESCRIPTION
**Pull request by AlexKlimaj**
This PR adds a hwdef for the ARK RTK GPS.

I am able to flash this AP_Periph binary through the dronecan gui and the PX4 cannode bootloader on the ARK RTK GPS will boot it.

Relevant issue when using two.
https://github.com/ArduPilot/ardupilot/issues/22253

### Procedure for fully updating the ARK RTK GPS from PX4 App and Bootloader

1. Enable Ardupilot CAN port
2. Connect one ARK RTK GPS to ardupilot CAN port
3. Open the dronecan gui and select slcan (Make sure the CAN port you are using corresponds with the FC SLCAN port)
4. Double click on the ARK RTK GPS node
5. Update firmware with the AP_Periph binary
6. When it is finished there will be a new node named org.ardupilot.ARK_RTK_GPS
7. To now update to the AP_Pheriph bootloader, open the node page
8. Fetch all params
9. Set FLASH_BOOTLOADER to 1
10. Verify in the logged messages that it says "Flash bootloader OK"
11. The ARK RTK GPS is now running AP Pheriph app and bootloader

https://arkelectron.com/product/ark-rtk-gps/

SAM_0882 (Large)

### CODE REVIEW

File: libraries/AP_HAL_ChibiOS/hwdef/ARK_RTK_GPS/hwdef.dat [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+define GPS_MAX_RECEIVERS 1
+define GPS_MAX_INSTANCES 1
+define HAL_COMPASS_MAX_SENSORS 1
+define HAL_PERIPH_ENABLE_GPS
+define HAL_PERIPH_ENABLE_MAG
+define HAL_PERIPH_ENABLE_BARO
+define HAL_PERIPH_ENABLE_BUZZER
+define HAL_PERIPH_ENABLE_RC_OUT
+define HAL_PERIPH_ENABLE_NOTIFY
```
Comment (peterbarker) [inline libraries/AP_HAL_ChibiOS/hwdef/ARK_RTK_GPS/hwdef.dat:133]: It really has all these or the facility to add them, then?

Comment (AlexKlimaj) [reply libraries/AP_HAL_ChibiOS/hwdef/ARK_RTK_GPS/hwdef.dat:133]: Yeah the only thing I'm unsure about is HAL_PERIPH_ENABLE_RC_OUT. Is that required for the buzzer?

Comment (peterbarker) [reply libraries/AP_HAL_ChibiOS/hwdef/ARK_RTK_GPS/hwdef.dat:133]: How is the buzzer connected?

Comment (peterbarker) [reply libraries/AP_HAL_ChibiOS/hwdef/ARK_RTK_GPS/hwdef.dat:133]: .... also, you could just remove it and test :-)

Comment (AlexKlimaj) [reply libraries/AP_HAL_ChibiOS/hwdef/ARK_RTK_GPS/hwdef.dat:133]: https://github.com/ARK-Electronics/ARK_RTK_GPS/blob/main/Hardware/Rev%203/ARK%20RTK%20GPS%20Rev%203%20Schematic.PDF

**Review by tridge — CHANGES_REQUESTED** [review]
we should discuss the DNA issue - the captures look like its giving 0,0,0,0 as the unique ID from px4 bootloader

**Comment by AlexKlimaj** [comment]
> 

@tridge its already set here https://github.com/ArduPilot/ardupilot/pull/22227/files#diff-1f4fe41225b3cbabbec89893b5b15bf9111a99fe4254b00fa7cd269de654e77aR118

File: libraries/AP_HAL_ChibiOS/hwdef/ARK_RTK_GPS/hwdef.dat (hunk 1 above)
Comment (peterbarker) [reply libraries/AP_HAL_ChibiOS/hwdef/ARK_RTK_GPS/hwdef.dat:133]: I think you can remove the RC Out - unless you have either servo outputs you want to drive or fancy serial-leds here somewhere.

**Comment by tridge** [comment]
@AlexKlimaj the comments on the PR need fixing, so "hwdef: xxxx" for hwdef changes, etc

**Comment by tridge** [comment]
ping me on discord voice sometime to discuss the DNA issue

**Comment by AlexKlimaj** [comment]
@tridge I think this is ready. @hendjoshsr71 has two ARK RTK GPS and can double check.

**Comment by hendjoshsr71** [comment]
I'm not available to test till after Christmas. I think this can merged though as Alex has tested it and a partner using this firmware has tested it (including gos for yaw).

**Review by hendjoshsr71 — APPROVED** [review]
Approve based upon partner testing

**Review by tridge — APPROVED** [review]

**Pull request merged**